src/hooks/Guilds/erc20/useERC20Balance.ts

```diff
@@ -0,0 +1,20 @@
+import ERC20 from 'contracts/ERC20.json';
+import { BigNumber } from 'ethers';
+import { useContractRead } from 'wagmi';
+
+export const useERC20Balance = (
+  contractAddress: string,
+  walletAddress: string
+) => {
+  const { data, isError, isLoading } = useContractRead({
+    addressOrName: contractAddress,
+    contractInterface: ERC20.abi,
+    functionName: 'balanceOf',
+    args: [walletAddress],

```

Comment:
We should probably add a watch here as well since we rely on fairly constant updates here